File: min-stack.cs
public class MinStack {
    private List<int> list;
    /** initialize your data structure here. */
    public MinStack() {
       list  = new List<int>();
    }
    
    public void Push(int x) {
        list.Add(x);
    }
    
    public void Pop() {
        list.RemoveAt(list.Count - 1);
    }
    
    public int Top() {
        return list[list.Count - 1];
    }
    
    public int GetMin() {
        return list.Min();
    }
}

/**
 * Your MinStack object will be instantiated and called as such:
 * MinStack obj = new MinStack();
 * obj.Push(x);
 * obj.Pop();
 * int param_3 = obj.Top();
 * int param_4 = obj.GetMin();
 */

File: 30-day-leetcoding-challenge/MinStack.java
class MinStack {

    private List<Integer> stack;
    private List<Integer> minStack;
    /** initialize your data structure here. */
    public MinStack() {
        stack = new LinkedList<Integer>();
        minStack = new LinkedList<Integer>();
    }
    
    public void push(int x) {
        stack.add(x);
        if(this.minStack.isEmpty() || x <= this.getMin())
            minStack.add(x);
    }
    
    public void pop() {
        if(this.getMin() == this.top())
            minStack.remove(minStack.size()-1);
        stack.remove(stack.size()-1);
    }
    
    public int top() {
        return stack.get(stack.size()-1);
    }
    
    public int getMin() {
        return minStack.get(minStack.size()-1);
    }
}
